tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,698 @@
+#
+# * this section of data is expected to be dominated by responses related to
+#   the stimulus onset, and we are not interested in this.
+#
+# In MNE we call plain FFT as a special case of Welch's method, with only a
+# single Welch window spanning the entire trial and no specific windowing
+# function (i.e. applying a boxcar window).
+#
+
+tmin = 1.
+tmax = 20.
+fmin = 1.
+fmax = 90.
+sf = epochs.info['sfreq']
+
+psds, freqs = mne.time_frequency.psd_welch(
+    epochs,
+    n_fft=int(sf * (tmax - tmin)),
+    n_overlap=0, n_per_seg=None,
+    tmin=tmin, tmax=tmax,
+    fmin=fmin, fmax=fmax,
+    window='boxcar',
+    verbose=False)
+
+
+###############################################################################
+# Calculate signal to noise ratio (SNR)
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# SNR - as we define it here - is a measure of relative power:
+# it's the ratio of power in a given frequency bin - the 'signal' -
+# to a 'noise' baseline - the average power in the surrounding frequency bins.
+# This approach was initially proposed by
+# `Meigen & Bach (1999) <https://doi.org/10.1023/A:1002097208337>`_
+#
+# Hence, we need to set some parameters for this baseline - how many
+# neighboring bins should be taken for this computation, and do we want to skip
+# the direct neighbors (this can make sense if the stimulation frequency is not
+# super constant, or frequency bands are very narrow).
+#
+# The function below does what we want.
+#
+
+def snr_spectrum(psd, noise_n_neighborfreqs=1, noise_skip_neighborfreqs=1):
+    """Compute SNR spectrum from PSD spectrum using convolution.
+
+    Parameters
+    ----------
+    psd : ndarray, shape=(n_trials, [n_channels,] n_frequency_bins)
+        containing psd values as spit out by MNE's PSD functions.
+        must be 2d or 3d array with frequencies in the last dimension
+    noise_n_neighborfreqs : int
+        number of neighboring frequencies used to compute noise level.
+        increment by one to add one frequency bin ON BOTH SIDES
+    noise_skip_neighborfreqs - int
+        set this >=1 if you want to exclude the immediately neighboring
+        frequency bins in noise level calculation
+
+    Returns
+    -------
+    snr - np.array
+        array containing snr for all epochs, channels, frequency bins.
+        NaN for frequencies on the edge, that do not have enoug neighbors on
+        one side to calculate snr
+    """
+    # Construct a kernel that calculates the mean of the neighboring
+    # frequencies
+    averaging_kernel = np.concatenate((
+        np.ones(noise_n_neighborfreqs),
+        np.zeros(2 * noise_skip_neighborfreqs + 1),
+        np.ones(noise_n_neighborfreqs)))
+    averaging_kernel /= averaging_kernel.sum()
+
+    # Calculate the mean of the neighboring frequencies by convolving with the
+    # averaging kernel.
+    mean_noise = np.apply_along_axis(
+        lambda psd_: np.convolve(psd_, averaging_kernel, mode='valid'),
+        axis=-1, arr=psd
```

Comment:
well, you can pass this function a 1d PSD array / a list (single channel, singel trial data) 
i guess this would create problems with `convolve2d`?

Reply:
`convolve2d` won't work with 3D-arrays, unfortunately. We could use `scipy.ndimage.filters.convolve` or `scipy.ndimage.filters.correlate` but they don't have the `mode='valid'` option which further complicates the code.